Added MemoryMap

EkkehardDomning opened the pull request:
Have look to this update. 
It implements an MemoryMap to allow the usage of optional memory and a simpler "case"-Management on setting memory and getting values from the memory.
On the long end, there is one method, that "insert" the optional 3K RAM, which funny enought leads to the display that the whole implementation has, before I wrote a single line. But it allows now to simulate the default "3583 Bytes free".

I had a short look into the new c++ code of the 6502 emulation. The changes are not direct compatible, since the method "step" is replaced by "run" etc. I am not sure if I find time to go through this implementation.

I am surprised that the high-resolution graphic features are not implemented, as well that the character ROM is not used. But maybe that beyond what the original idea behind this small project was.

Greetings Ekkehard

EkkehardDomning committed:
Added MemoryMap

Dennis1000 commented:
The project is a MOS 6502 emulation implementation - the C64 and VIC-20 "emulation" is just a quick "side-hack" and - of course - far from complete ;-)

EkkehardDomning commented:
> The project ist a MOS 6502 emulation implementation - the C64 and VIC-20 "emulation" is just a quick "side-hack" and - of course - far from complete ;-)

Sure, I was aware of this. The VICE-Project is much closer too perfection, but it is not Pascal/Delphi/Lazarus :-)
Great work anyway!

EkkehardDomning committed:
Implemented changes made in the C++ implementation for the MOS Technology 6502 CPU by Gianluca Ghettini.
Updated TestSuite for Lazarus
Updated C64 for Lazarus (still not running with Lazarus)
Updated VIC-20

EkkehardDomning committed:
Bugfix, compatibility Memory changed from pointer to array of byte

Dennis1000 commented:
Thanks for you PRs, keep 'em coming :-)
